pinot-core/src/main/java/com/linkedin/pinot/core/minion/rollup/RollupRecordAggregator.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (C) 2014-2018 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.minion.rollup;
+
+import com.linkedin.pinot.common.data.MetricFieldSpec;
+import com.linkedin.pinot.common.data.Schema;
+import com.linkedin.pinot.core.data.GenericRow;
+import com.linkedin.pinot.core.minion.rollup.aggregate.ValueAggregator;
+import com.linkedin.pinot.core.minion.rollup.aggregate.ValueAggregatorFactory;
+import com.linkedin.pinot.core.minion.segment.RecordAggregator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Record aggregator implementation for roll-up segment converter.
+ *
+ * Given the list of rows with the same dimension values, the record aggregator aggregates rows into a single row
+ * whose metric column values are aggregated based on the given aggregator functions.
+ */
+public class RollupRecordAggregator implements RecordAggregator {
+  private final static ValueAggregator DEFAULT_AGGREGATOR_FUNCTION = ValueAggregatorFactory.getValueAggregator("SUM");
```

Comment:
It'd be better to use `ValueAggregatorType` instead of a String "SUM" here?

Reply:
Otherwise, I will need to put `ValueAggregatorFactory.ValueAggregatorType.SUM.name()` which is much longer than `SUM` string. How do you think?